src/randomInt.test-d.ts

```diff
@@ -0,0 +1,64 @@
+import { randomInt } from "./randomInt";
+
+test("returns number when from is float", () => {
+  expectTypeOf(randomInt(0.5, 1)).toEqualTypeOf<number>();
+});
+
+test("returns number when to is float", () => {
+  expectTypeOf(randomInt(1, 1.5)).toEqualTypeOf<number>();
+});
+
+test("returns never when from is greater than to", () => {
+  expectTypeOf(randomInt(10, 1)).toEqualTypeOf<never>();
+});
+
+test("returns the range as union for valid range (0-999)", () => {
+  expectTypeOf(randomInt(0, 5)).toEqualTypeOf<0 | 1 | 2 | 3 | 4 | 5>();
+  expectTypeOf(randomInt(998, 999)).toEqualTypeOf<998 | 999>();
+});
+
+test("returns the same value when to equals from", () => {
+  expectTypeOf(randomInt(1, 1)).toEqualTypeOf<1>();
+});
+
+test("returns the same value when to equals from and out of range", () => {
+  expectTypeOf(randomInt(10_000, 10_000)).toEqualTypeOf<10_000>();
+});
+
+test("Returns number for invalid range (to >= 1000)", () => {
```

Comment:
the range isn't invalid, just out of range for our more refined type.
```suggestion
test("Returns number for large ranges (to >= 1000)", () => {
```